src/GafferImage/Ramp.cpp

```diff
@@ -0,0 +1,280 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2018, Lucien Fostier. All rights reserved.
+//  Copyright (c) 2012-2018, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/Transform2DPlug.h"
+#include "Gaffer/Context.h"
+
+#include "GafferImage/Ramp.h"
+#include "GafferImage/ImageAlgo.h"
+
+using namespace std;
+using namespace Imath;
+using namespace IECore;
+using namespace GafferImage;
+using namespace Gaffer;
+
+//////////////////////////////////////////////////////////////////////////
+// Ramp implementation
+//////////////////////////////////////////////////////////////////////////
+
+IE_CORE_DEFINERUNTIMETYPED( Ramp );
+
+size_t Ramp::g_firstPlugIndex = 0;
+
+Ramp::Ramp( const std::string &name )
+	:	ImageNode( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new FormatPlug( "format" ) );
+	addChild( new V2fPlug( "positionA", Plug::In ) );
+	addChild( new V2fPlug( "positionB", Plug::In ) );
+	SplinefColor4fPlug::ValueType rampDefault;
```

Comment:
This defaults to CatmullRom. I think there have been some discussions lately about MontoneCubic maybe being a better default? @danieldresser, any opinions?